src/lib/near-fastauth-wallet.ts

```diff
@@ -400,6 +402,26 @@ const FastAuthWallet: WalletBehaviourFactory<
     resetRelayerUrl() {
       relayerUrl = params.relayerUrl;
     },
+    async signMessage(data) {
+      return _state.wallet.requestSignMessage(data);
+    },
+    async verifyMessageSignature(
+      message: SignMessageParams,
+      signedMessage: SignedMessage
+    ): Promise<boolean> {
+      const accessKeys = await _state.wallet.account().getAccessKeys();
+      const isFullAccessKey = accessKeys.some(
+        (key) =>
+          key.public_key === signedMessage.publicKey.toString() &&
+          key.access_key.permission === 'FullAccess'
+      );
+
+      if (!isFullAccessKey) {
+        return false;
```

Comment:
Good point, this method it's not on the NEP but I added it to make it easy to verify the `signMessage` signature.

For the `signMessage`, the signature must be done by a FullAccess Key as state on the NEP: https://github.com/near/NEPs/blob/master/neps/nep-0413.md#why-using-a-fullaccess-key-why-not-simply-creating-an-functioncall-key-for-signing

What do you think? Should we include this method on the wallet?

Reply:
Yes, its reasonable to avoid such attack vector. 

>> Should we include this method on the wallet?
could you elaborate please? Which wallet do you mean?

Btw, for `verifyMessageSignature` method, do you need to verify if `signedMessage.publicKey` belongs to to `_state.wallet` keys?

Reply:
> Could you elaborate please? Which wallet do you mean?

I mean, the method `verifySignMessage` isn't included on the NEP. I added it to `near-fastauth-wallet`, so the integrator don't need to implement it's own. It can use it to verify if the `SignedMessage` it's valid. 

> Btw, for verifyMessageSignature method, do you need to verify if signedMessage.publicKey belongs to to _state.wallet keys?

It's not clear to me why the `publicKey` should be included on the `_state.wallet`. Can you elaborate more? 

By what I understand, the message can be signed by any `FullAccess Key` of the account, it doesn't need to be available on the `_wallet.state`. 

On this piece of code:

```
const accessKeys = await _state.wallet.account().getAccessKeys();
      const isFullAccessKey = accessKeys.some(
        (key) =>
          key.public_key === signedMessage.publicKey.toString() &&
          key.access_key.permission === 'FullAccess'
      );
```

We validate that the `signedMessage.publicKey` it's a `FullAccess Key` and it also belongs to the account in the wallet.

Reply:
>>  I mean, the method verifySignMessage isn't included on the NEP. I added it to near-fastauth-wallet, so the integrator don't need to implement it's own. It can use it to verify if the SignedMessage it's valid.

Yes, it is worth adding your code to the near-api-js, at the very least.